### ISSUE TITLE
Pyright doesn't allow generic explicit TypeAlias

### ISSUE THREAD

**Issue by thomkeh** [microsoft/pyright#2834]
**Describe the bug**
When using `TypeAlias` with a type that is generic, pyright is unhappy.

**To Reproduce**
```python
from typing import Generic, TypeAlias, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    def __init__(self, a: T):
        self._a = a

B: TypeAlias = A

x: B[int]
x = A(a=3)
```

pyright output
```
[...]/alias.py
  [...]/alias.py:11:4 - error: Type "A[Unknown]" is already specialized
1 error, 0 warnings, 0 infos
```
When I run it in strict mode, I get the additional warning:
```
alias.py:9:1 - error: Type of "B" is partially unknown
```

When I replace `B: TypeAlias` with just `B`, then there is no error.

As a side note, mypy 0.931 doesn't show an error even with `TypeAlias`.

**Expected behavior**  
I think the given code should be fine, though [PEP 613](https://www.python.org/dev/peps/pep-0613/) doesn't mention it explicitly. Mypy also seems to think it's fine.

**Screenshots or Code**
(see above)

**VS Code extension or command-line**
CLI, version 1.1.205

**Comment by erictraut** [comment]
If your intent is for the type alias `B` to be generic, you should include a TypeVar for a type argument, like this:
```python
B: TypeAlias = A[T]
```

Because you have left off the type argument, it is interpreted as `A[Any]`.

We recently had a discussion about this on the [python/typing discussion forum](https://github.com/python/typing/discussions/1011) . We concluded there that for implicit type aliases (those that don't use `TypeAlias`, we would be a bit more lax and allow `B = A` to not be interpreted as `A[Any]`. That makes sense because `B = A` could be interpreted as a normal variable assignment rather than a type alias definition. However, with an _explicit_ type alias declaration, the type argument should be provided so it is clear whether you mean `A[Any]` or `A[T]`. By default, pyright assumes `A[Any]` in this case.

**Comment by erictraut** [comment]
Also, you might want to enable the `reportMissingTypeArgument` diagnostic check so pyright warns you when you have omitted a type argument on a generic class or type alias.

**Comment by thomkeh** [comment]
I see, thanks!

**Comment by thomkeh** [comment]
@erictraut it is a bit sad though that `isinstance()` breaks when you explicitly give `TypeVar`s (at least in Python 3.9+):

```
Python 3.10.0 (default, Dec 21 2021, 13:36:04) [GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> G = set | list
>>> isinstance([3], G)
True
>>> from typing import TypeAlias, TypeVar
>>> T = TypeVar("T")
>>> H: TypeAlias = set[T] | list[T]
>>> isinstance([3], H)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() argument 2 cannot contain a parameterized generic
```
